### PR TITLE
PUB-2499 - Show error notification when Instagram disconnect occurs

### PR DESCRIPTION
**Pull request by karelm**
## Description

Show error notification when Instagram disconnect occurs
[PUB-2499](https://buffer.atlassian.net/browse/PUB-2499)
## Checklist

-   [ ] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] I have tested different user stories. 
-   [ ] I have identified similar or related features and tested they are still working.
-   [ ] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [ ] I have commented my code, particularly in hard-to-understand areas.
-   [ ] I kept accessibility in mind by following the [a11y checklist.](https://www.notion.so/buffer/Workflow-Checklist-e64d86eb795140bcbfdc16d1c72e573f)

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: packages/profiles-disconnected-banner/components/ProfilesDisconnectedBanner/index.jsx [hunk 1]
```diff
@@ -8,23 +8,34 @@ const TextWithStyles = styled(Text)`
   margin: 0;
 `;
 
+const ExtraMessageWithStyles = styled(Text)`
+  margin: 0;
+  font-weight: bold;
```
Comment (anafilipadealmeida) [inline packages/profiles-disconnected-banner/components/ProfilesDisconnectedBanner/index.jsx]: Another way of doing this is to `import { fontWeightBold } from '@bufferapp/ui/style/fonts';` and use it 🙂

File: packages/profiles-disconnected-banner/index.js [hunk 2]
```diff
@@ -3,13 +3,26 @@ import { actions as profilesDisconnectedModalActions } from '@bufferapp/publish-
...
-  state => ({
-    profileId: state.profileSidebar.selectedProfileId,
-    service:
-      state.profileSidebar.selectedProfile &&
-      state.profileSidebar.selectedProfile.service,
-    translations: state.i18n.translations['profiles-disconnected-banner'],
-  }),
+  state => {
+    let extraMessage = null;
+    const selectedProfile = state?.profileSidebar?.selectedProfile;
```
Comment (anafilipadealmeida) [inline packages/profiles-disconnected-banner/index.js]: 😍 😍

File: packages/profiles-disconnected-banner/index.js [hunk 3]
```diff
@@ -3,13 +3,26 @@ import { actions as profilesDisconnectedModalActions } from '@bufferapp/publish-
...
+
+    if (
+      selectedProfile?.service === 'instagram' &&
+      selectedProfile?.service_type === 'profile'
+    ) {
+      extraMessage = translations?.extraMessage?.instagram;
+    }
+
+    return {
+      profileId: state.profileSidebar.selectedProfileId,
```
Comment (anafilipadealmeida) [inline packages/profiles-disconnected-banner/index.js]: How about replacing it by the const selectedProfile?

File: packages/profiles-disconnected-modal/components/ProfilesDisconnectedModal/index.jsx [hunk 4]
```diff
@@ -4,33 +4,63 @@ import PropTypes from 'prop-types';
...
+  margin: 0;
+  font-weight: bold;
+`;
+
+const DisconnectedProfile = styled.div`
+  display: flex;
+  justify-content: space-between;
+  border: 1px solid #e0e0e0;
+  padding: 8px;
+  border-radius: 4px;
```
Comment (anafilipadealmeida) [inline packages/profiles-disconnected-modal/components/ProfilesDisconnectedModal/index.jsx]: import { borderRadius } from '@bufferapp/ui/style/borders';

File: packages/profiles-disconnected-modal/components/ProfilesDisconnectedModal/index.jsx [hunk 5]
```diff
@@ -4,33 +4,63 @@ import PropTypes from 'prop-types';
...
+
+const ExtraMessageWithStyles = styled(Text)`
+  margin: 0;
+  font-weight: bold;
+`;
+
+const DisconnectedProfile = styled.div`
+  display: flex;
+  justify-content: space-between;
+  border: 1px solid #e0e0e0;
```
Comment (anafilipadealmeida) [inline packages/profiles-disconnected-modal/components/ProfilesDisconnectedModal/index.jsx]: import { grayLight } from '@bufferapp/ui/style/colors';

File: packages/profiles-disconnected-modal/components/ProfilesDisconnectedModal/index.jsx [hunk 6]
```diff
@@ -39,10 +69,10 @@ const ProfilesDisconnectedModal = ({
                   type="social"
                   network={p.service}
                 />
-                <div style={{ marginLeft: '16px' }}>
+                <AvatarName>
                   <Text weight="medium">{p.formatted_username}</Text>
```
Comment (anafilipadealmeida) [inline packages/profiles-disconnected-modal/components/ProfilesDisconnectedModal/index.jsx]: If this Text is already from the bufferapp/ui, I don't think this `weight` attribute is doing anything 🤔 I think the attribute only exists in bufferapp/components

Comment (karelm) [reply packages/profiles-disconnected-modal/components/ProfilesDisconnectedModal/index.jsx]: Thanks for pointing that out.. I did not know that was the case, so I'm very grateful 🙌

**Comment by karelm** [comment]
> Thank you so much for doing this @karelm! Just left a few small comments! And I loved the optional chaining 😍

Thanks Ana, I was so focussed on making the wrappers semantically correct that I forgot to check the contents of them. So I appreciate you highlighting font colours, weights, and border variables 🥇 Thanks so much for your input

**Comment by karelm** [comment]
> Love this @karelm and ➕1 to @anafilipadealmeida's great UI feedback.
> 
> One small nitpick, is it hard to decreates the text padding on the error box in the queue? Would love if it didn't wrap quite as much and looked more like this:
> image

Thanks @hamstu for pointing that out, I've pushed up a fix to use the whole space. I didn't really stop to think about how large an area wasn't being used... 150x was definitely a lot, so I've made an adjustment that should be fine most of the time.

**Comment by hamstu** [comment]
Thanks so much @karelm! That's perfect. 👌

**Pull request merged**